python/ray/train/batch_predictor.py

```diff
@@ -179,6 +181,13 @@ def predict(
             )
             predictor_kwargs["use_gpu"] = True
 
+        # For DL predictors that implemented _predict_numpy, we prefer to use
+        # numpy batch format for better perf and user experience.
+        has_predict_numpy = (
+            self._predictor_cls._predict_numpy != Predictor._predict_numpy
+        )
+        batch_format = BatchFormat.NUMPY if has_predict_numpy else BatchFormat.PANDAS
```

Comment:
I think we can remove this check here and do it only inside `Predictor`.

On the `BatchPredictor` side we can just do something like in this PR: https://github.com/ray-project/ray/pull/29088/files

```
dataset_format = dataset._dataset_format()
if dataset_format not in ("pandas", "arrow"):
    raise ValueError(
        f"Unsupported Dataset format: '{dataset_format}'. Only 'pandas' "
        "and 'arrow' Dataset formats are supported."
    )

return dataset.map_batches(self._transform_batch, batch_format=dataset_format)
```

we just map over the native format, and then delegate to the Predictor to do the necessary conversions and routing.

Reply:
This way conversion logic can be contained inside `Predictor` and doesn't need to bubble up to higher level APIs like `BatchPredictor`

Reply:
it makes sense at high level but our `BatchPredictor` essentially just a call to `map_batches` with a hardcoded `batch_format` , same for the `ScoringWrapper` class, we need to provide a value for them at `BatchPredictor` level that's not pandas for Torch/Tensorflow predictor right ?

Reply:
I see.

The problem with the current implementation though is the preprocessor may have a pandas only path right?

so then we'll be doing pandas -> [map_batches]-> numpy ->[preprocessor]-> pandas ->[predictor]-> numpy.

Reply:
discussed offline, added comment about why we need to decide and call `map_batches` with right format early